### PR TITLE
feat(parser): parse decimal type

### PR DESCRIPTION
**Pull request by andylokandy**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

Summary about this PR

Related to https://github.com/datafuselabs/databend/issues/2931

### CODE REVIEW

File: src/query/ast/src/parser/expr.rs [hunk 1]
```diff
@@ -1110,11 +1110,17 @@ pub fn type_name(i: Input) -> IResult<TypeName> {
...
     let ty_float32 = value(TypeName::Float32, rule! { FLOAT32 | FLOAT });
     let ty_float64 = value(TypeName::Float64, rule! { FLOAT64 | DOUBLE });
-    let ty_decimal = map(
+    let ty_decimal = map_res(
         rule! { DECIMAL ~ "(" ~ #literal_u64 ~ "," ~ #literal_u64 ~ ")" },
-        |(_, _, precision, _, scale, _)| TypeName::Decimal {
-            precision: precision as u8,
-            scale: scale as u8,
+        |(_, _, precision, _, scale, _)| {
+            Ok(TypeName::Decimal {
```
Comment (sundy-li) [inline src/query/ast/src/parser/expr.rs:1116]: No validation here ？

Comment (andylokandy) [reply src/query/ast/src/parser/expr.rs:1116]: the precise validation should be in DecimalDataType::from_size()

**Review by sundy-li — APPROVED** [review]

**Comment by BohuTANG** [comment]
```
0: statement is expected to fail with error:
	1005
but got error:
	mysql client error: Input/output error: Input/output error: connection closed
[SQL] CREATE TABLE test_num_data (id integer, val decimal(18,10))
at tests/sqllogictests/suites/duckdb/issues/monetdb/test_round_bug.test:1
```

**Comment by andylokandy** [comment]
@mergify update

**Comment by andylokandy** [comment]
@sundy-li PTAL

**Pull request merged**